scripts/kconfig/merge_config.py

```diff
@@ -3,50 +3,91 @@
 import argparse
 import re
 
+
 def arguments_parse():
     parser = argparse.ArgumentParser()
     parser.add_argument('-q', dest="quiet", action='store_true')
-    parser.add_argument('-m', dest='runmake', action='store_false',
-        help='only merge the fragments, do not execute the make command')
-    parser.add_argument('-n', dest='alltarget', action='store_const', const='allnoconfig', default='alldefconfig',
-        help='use allnoconfig instead of alldefconfig')
     parser.add_argument('-r', dest='warnredun', action='store_true',
-        help='list redundant entries when merging fragments')
+                        help='list redundant entries when merging fragments')
     parser.add_argument('-O', dest='output_dir',
-        help='to put generated output files', default='.')
+                        help='to put generated output files', default='.')
     parser.add_argument('config', nargs='+')
 
     return parser.parse_args()
 
 
+def get_config_name(line):
+    # '# CONFIG_FOO is not set' should be treated by merge_config as a
+    # state like any other
+
+    is_not_set_pattern = re.compile('^# (CONFIG_[a-zA-Z0-9_]*) is not set.*')
+    pattern            = re.compile('^(CONFIG_[a-zA-Z0-9_]*)[= ].*')
```

Comment:
> "It matches strings like "CONFIG_FOO did not actually receive a value"

That is a minor problem because that is invalid syntax, but it would be great for
usability if you were able to fix this!

> If the base file sets "# CONFIG_FOO is not set", and some other fragment sets "CONFIG_FOO=n", that gets treated like an override, even though it's redundant (and CONFIG_FOO=n should be changed to "is not set").

I don't see how this affects the user, but if you can find a clean fix then why not :)

Reply:
> That is a minor problem because that is invalid syntax, but it would be great for
> usability if you were able to fix this!

Glad you agree it's OK to fix. It's fixed in my patch.

> I don't see how this affects the user, but if you can find a clean fix then why not :)

This script prints warnings on overrides. Warnings affect users.